### PR TITLE
Add join page, changed navbar on all pages

### PR DESCRIPTION
**Pull request by usephysics**
Added a new page at http://waterloorocketry.com/join. Has information and Q&As for students wanting to join the team.

Changed navbar on every page to include the join page - I put it at the front.

Also, changed my own title from "Member" to "Software"

---
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website/116)

### CODE REVIEW

File: join.html [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                <div class="col-10 col-md-8 col-lg-7 mx-auto answer collapse" id="four">
+                    <div class="answer-text">
+                        What you start off with will depend on what stage our projects are at, as well as your own preference and comfort level. 
+                        There is usually a variety of ongoing tasks to choose from. Some examples of past projects that first-years have worked on are:
+                        <br>- Designing a test operations procedures simulator in Python
+                        <br>- Sourcing and obtaining sponsorship for components of our data acquisition system
+                        <br>- Designing part of a scientific experiment measuring radiation throughout flight
+                        <br>- Machining an aluminium bulkhead for the recovery system
+                        <br>- Representing the team at high school and university outreach events
+                        </ul>
```
Comment (usephysics) [inline join.html:123]: Oops, this shouldn't be here

**Review by wendi-yu — COMMENTED** [review]
This page was a great idea, I really like it! I poked the page on Chrome, Safari, and Edge and they all seem to work well. The mobile view also looks good.

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website/116#-:-MEaIc_X16ujt99Jp5-b:bnj9me5)* status: 0 of 20 files reviewed, 12 unresolved discussions (waiting on @jacobdeery, @usephysics, and @wendi-yu)

___
*[join.html, line 55 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaCQYI57drxL2Gr5ik:-MEaCQYJDfIzjJj2wzM4:bjgeu2g) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L55)):*
> ```HTML
>                     <h1 class="content-title">HOW DO I JOIN?</h1>
>                     <p class="content-text">
>                         If you haven't already, join our <a href="https://www.facebook.com/groups/uwrocketry/">Facebook group</a>
> ```

I'd suggest a `target=_blank` on the link here so that it opens in a new tab and doesn't redirect them away from the join page :0
___
*[join.html, line 65 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaCx-c3NjZ3jBP4fC4:-MEaCx-c3NjZ3jBP4fC5:b-6ymp2e) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L65)):*
> ```HTML
>             <div class="row my-md-3 my-2">
>                 <div class="col-12">
>                     <h1 class="content-title text-center">FREQUENTLY ASKED QUESTIONS</h1>
> ```

This is entirely me being picky, but I think the UI for the FAQ drop down bars might look nicer if we 
* make the bar colour a lighter grey
* maybe round the edges?
* either have the FAQ section wider, or the top text narrower (so they're more similar in width). 

Happy to take any input others might have on this, UI isn't my specialty this is just my gut feeling lol. It's super cool that you made the toggling work though, that sounds like a nightmare and a half to put together. Works well :)
___
*[join.html, line 70 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaIkUmAS-v18Grp_fs:-MEaIkUmAS-v18Grp_ft:bmxbukt) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L70)):*
> ```HTML
>             <div class="row my-0 faq-container">
>                 <div class="col-10 col-md-8 col-lg-7 px-0 mx-auto">
>                     <button class="btn text-left" type="button" data-toggle="collapse" data-target="#one" aria-expanded="false">
> ```

A UI thing I noticed was that when you click the toggle question button, it stays highlighted until you click elsewhere. Not a blocker I don't think, but it would be nice if it didn't stay highlighted :)
___
*[join.html, line 95 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaED9D9a26qNZ1ofgp:-MEaED9D9a26qNZ1ofgq:b-6jgoyn) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L95)):*
> ```HTML
>                 <div class="col-10 col-md-8 col-lg-7 px-0 mx-auto">
>                     <button class="btn text-left" type="button" data-toggle="collapse" data-target="#three" aria-expanded="false">
>                         What does the team's day to day operations look like?
> ```

What does -> What do (since operations is plural I think?)
___
*[join.html, line 101 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaEK4x85NXyYP07ijv:-MEaEK4x85NXyYP07ijw:b-c5w8j8) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L101)):*
> ```HTML
>                     <div class="answer-text">
>                         The team is split up into different subteams - for example - electrical, recovery, propulsion, etc. Each
>                         subteam was regular meetings, usually once a week, while the entire team gathers for a general meeting once
> ```

was -> has
Also "general meeting once a week" -> "weekly general meeting" (we probably want to avoid repeating "once a week"?)
___
*[join.html, line 102 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaEMfY4G832WQpAcBn:-MEaEMfY4G832WQpAcBo:b-nxn7pf) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L102)):*
> ```HTML
>                         The team is split up into different subteams - for example - electrical, recovery, propulsion, etc. Each
>                         subteam was regular meetings, usually once a week, while the entire team gathers for a general meeting once
>                         per week. Each subteam is responsible for a different part of the rocket and it's supporting equipment;
> ```

it's -> its
___
*[join.html, line 103 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaEPTYDNB1-cH6WqUL:-MEaEPTYDNB1-cH6WqUM:b-bb9oap) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L103)):*
> ```HTML
>                         subteam was regular meetings, usually once a week, while the entire team gathers for a general meeting once
>                         per week. Each subteam is responsible for a different part of the rocket and it's supporting equipment;
>                         it's operation supervised by a subteam lead. You are not restricted to any one subteam; you can be apart of as 
> ```

how about "... part of the rocket and its supporting equipment, and its operations are supervised by a subteam lead."? There's another semicolon next sentence (being picky here again lol)
___
*[join.html, line 104 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaI3hMB6T3DAF8Dhm4:-MEaI3hMB6T3DAF8Dhm5:be35vwy) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L104)):*
> ```HTML
>                         per week. Each subteam is responsible for a different part of the rocket and it's supporting equipment;
>                         it's operation supervised by a subteam lead. You are not restricted to any one subteam; you can be apart of as 
>                         many as you want. Communication is done primarily through a Slack server.
> ```

I think Slack calls them workspaces, not servers
___
*[join.html, line 114 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaGixW3mmNQgnhYDI1:-MEaGixW3mmNQgnhYDI2:bis2lpg) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L114)):*
> ```HTML
>                 <div class="col-10 col-md-8 col-lg-7 mx-auto answer collapse" id="four">
>                     <div class="answer-text">
>                         What you start off with will depend on what stage our projects are at, as well as your own preference and comfort level. 
> ```

I think this could also be a great place to point to the "currently working on" sections on the subsystems page ;)
___
*[join.html, line 116 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaFZyDAyKqJdGIw-nr:-MEaFZyDAyKqJdGIw-ns:b-k1it1p) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L116)):*
> ```HTML
>                         What you start off with will depend on what stage our projects are at, as well as your own preference and comfort level. 
>                         There is usually a variety of ongoing tasks to choose from. Some examples of past projects that first-years have worked on are:
>                         <br>- Designing a test operations procedures simulator in Python
> ```

For the sake of consistency with other lists on the site I think we should use a `<ul>` element here (bullet points, essentially), instead of manual line breaks and dashes. You can do this just by popping the bullet point inside `<li>` tags and wrapping the whole list in a `<ul>`:
```
<ul> 
    <li>Designing a test operations procedures simulator in Python</li>
</ul>
```
you seem to have the closing `</ul>` tag already but not a matching starting one? You might also have to explicitly set the `list-style-type` property to `bullet` in the CSS for the `ul` element to get the bullets to show properly.
___
*[join.html, line 126 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaGrWf-jofBkbj5T5W:-MEaGrWf-jofBkbj5T5X:bnmehuj) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L126)):*
> ```HTML
>                 <div class="col-10 col-md-8 col-lg-7 px-0 mx-auto">
>                     <button class="btn text-left" type="button" data-toggle="collapse" data-target="#five" aria-expanded="false">
>                         Can I join Waterloo Rocketry, if I am in a different student design team?
> ```

No comma
___
*[join.html, line 131 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaGt_X6BDdv75fF9mj:-MEaGt_X6BDdv75fF9mk:briow9j) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L131)):*
> ```HTML
>                 <div class="col-10 col-md-8 col-lg-7 mx-auto answer collapse" id="five">
>                     <div class="answer-text">
>                         Sure! We don’t mind if you’re on another design team, but there’s no guarantee our meeting times won’t overlap.
> ```

Maybe gently mention something about managing time commitments?
___
*[css/join.css, line 10 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaKNSk8osTnec5DTyA:-MEaK0-r5u8RvdUff_bC:b-2mcmmi) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/css/join.css#L10)):*
> ```CSS
> 
> .faq-container {
>     font-family: 'Open Sans', Helvetica, sans-serif;
> ```

I think this font is the default in the site-wide CSS, isn't it? Does it work if you give it the `content-text` class then change just the `font-size` in the CSS?
___
*[css/join.css, line 34 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaJXht9lwpLIw5L6YR:-MEaJXht9lwpLIw5L6YS:bwpnh8n) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/css/join.css#L34)):*
> ```CSS
> .answer-text {
>     padding: 0.5rem 0;
> }
> ```

newline :0

**Comment by usephysics** [comment]
Thanks for the review Wendi! I've implemented most of your suggestions; here's a few comments:

> 
> > ```
> >                     <h1 class="content-title">HOW DO I JOIN?</h1>
> >                     <p class="content-text">
> >                         If you haven't already, join our <a href="https://www.facebook.com/groups/uwrocketry/">Facebook group</a>
> > ```
> 
> I'd suggest a `target=_blank` on the link here so that it opens in a new tab and doesn't redirect them away from the join page :0

Use of target=_blank is the coke vs pepsi equivalent of UX - the argument being, without using it, if the user wishes to open the page in a new tab, they have that choice. When using it, you've taken away choice from the user. I am on the side of not using it, but I'm not passionate about it - so if others think it's a good idea, that's fine with me


> > ```
> >             <div class="row my-0 faq-container">
> >                 <div class="col-10 col-md-8 col-lg-7 px-0 mx-auto">
> >                     <button class="btn text-left" type="button" data-toggle="collapse" data-target="#one" aria-expanded="false">
> > ```
> 
> A UI thing I noticed was that when you click the toggle question button, it stays highlighted until you click elsewhere. Not a blocker I don't think, but it would be nice if it didn't stay highlighted :)

The highlight is removable, but it shouldn't be due to accessibility issues (keyboard only users). See here: http://www.outlinenone.com/

> > ```
> >                         What you start off with will depend on what stage our projects are at, as well as your own preference and comfort level. 
> >                         There is usually a variety of ongoing tasks to choose from. Some examples of past projects that first-years have worked on are:
> >                         <br>- Designing a test operations procedures simulator in Python
> > ```
> 
> For the sake of consistency with other lists on the site I think we should use a `<ul>` element here (bullet points, essentially), instead of manual line breaks and dashes. You can do this just by popping the bullet point inside `<li>` tags and wrapping the whole list in a `<ul>`:
> 
> ```
> <ul> 
>     <li>Designing a test operations procedures simulator in Python</li>
> </ul>
> ```
> 

I tried this initially, but bootstrap wasn't cooperating

**Review by wendi-yu — COMMENTED** [review]
Awesome yeah, thanks for making the changes! I resolved the relevant stuff and left in the comments for a few that are left. I'm using reviewable for the reviews so the comments are directly replyable/resolvable here, but apparently reviewable comments don't show on the existing github review system lol. I can move to there if you prefer?

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website/116#-:-MEeEqpM1Q64mv2TmKRo:bnr4nr9)* status: 0 of 20 files reviewed, 6 unresolved discussions (waiting on @jacobdeery and @usephysics)

___
*[join.html, line 55 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaCQYI57drxL2Gr5ik:-MEeA1q0-kfswMcf1vyf:b-bkyowv) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L55)):*
> Use of target=_blank is the coke vs pepsi equivalent of UX - the argument being, without using it, if the user wishes to open the page in a new tab, they have that choice. When using it, you've taken away choice from the user. I am on the side of not using it, but I'm not passionate about it - so if others think it's a good idea, that's fine with me

Oh I honestly wasn't aware of this debate (I'm a fake web developer lol). This sounds fine by me, I really don't have an opinion at all. Might be worth going through and doing this consistently throughout the site though, which is a pretty chonky especially for sponsors, but doesn't have to happen in this PR :)
___
*[join.html, line 70 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaIkUmAS-v18Grp_fs:-MEeBX_QCf4cJEaMfzxu:bn6kiiy) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L70)):*
> The highlight is removable, but it shouldn't be due to accessibility issues (keyboard only users). See here: http://www.outlinenone.com/

I was _also_ unaware of this one, ty for the heads up :)
___
*[join.html, line 116 at r1](https://reviewable.io/reviews/waterloo-rocketry/website/116#-MEaFZyDAyKqJdGIw-nr:-MEeE7Zz-y5pvPBPByHA:b9r8lsl) ([raw file](https://github.com/waterloo-rocketry/website/blob/b870a6bd31f62612f26f17b66ab09ab4c298e98c/join.html#L116)):*
> I tried this initially, but bootstrap wasn't cooperating

I think I got a version of this working on my local copy, I can send you the code or add a commit that you're free to remove/modify/cherry-pick or whatever? I'm not sure what exactly on your end wasn't working, but I had some formatting issues I think were caused by the CSS on surrounding elements and that I just fixed manually with CSS rules for `ul` lol.

**Comment by usephysics** [comment]
@wendi-yu I've added the rest of what is unresolved on reviewable - I made the buttons a lighter grey but I checked rounding the FAQ and increasing its width and I don't think they look better

**Review by wendi-yu — APPROVED** [review]
Yeah that's totally fair, just a UI suggestion on my part. Looks good otherwise :) There's that newline still missing at the end of the `join.css` file but otherwise LGTM

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website/116#-:-MFEt6Oa0huFgXL9FGna:bdoxh51)* status: 0 of 20 files reviewed, 2 unresolved discussions (waiting on @jacobdeery and @usephysics)

**Pull request merged**